firestore-tos-extension/POSTINSTALL.md

```diff
@@ -0,0 +1,149 @@
+### See it in action
+
+You can test out this extension right away!
+
+1. Go to your [Authentication dashboard](https://console.firebase.google.com/project/${param:PROJECT_ID}/authentication/users) in the Firebase console.
+
+1. Click **Add User** to add a test user, then copy the test user's UID to your clipboard.
+
+2. Access the user through any of the authetication methods.
+
+```js
+   // v9
+   import { getAuth, onAuthStateChanged } from "firebase/auth";
+
+   const auth = getAuth();
+   const user = await auth.getUser(uid);
+```
+
+```js
+   // < v9
+   import firebase from "firebase-admin";
+
+   const user = await firebase.auth().getUser(uid);
+```
+
+3. Provide the user / userId into one of the available functions.
+
+### Using the extension
+
+All ackowledgement documents are managed in the Cloud Firestore database.
+
+These can be found in the `${param:COLLECTION_PATH}` collection.
+
+This will also contain two pre-defined subcollections:
+
+- acknowledgements
+- terms
+
+The extension will provide `4` new functions for managing ackowledgement for users.
+
+**Create Terms**
+
+  Creates a new terms of service, based on the following schema:
+
+  ```js
+    export interface TermsOfServiceMetadata {
+        tosId: string;
+        link: string;
+        creationDate: string;
+        allowList?: string[];
+        noticeType?: {
```

Comment:
Is this a mapping so that we can store the user preferences here? That seems slightly confusing to me. Maybe it makes more sense as a separate field? like preferences

Reply:
This is to support the roles (or any other custom data), although preferences is currently a value in here. We could separate preferences as a separate field?

Reply:
Yeah, i think we should separate preferences to a separate field. And have a "notice_type" string field